### PR TITLE
Wrap world

### PR DESCRIPTION
**Pull request by tallytalwar**
This is mostly working, but I am going to change the implementation logic a bit to better caching for wrapped tiles.

- [x] better use of tilecache for wrapped tiles.

### CODE REVIEW

**Comment by blair1618** [comment]
Reviewing

File: core/src/tile/tileManager.cpp [hunk 1]
```diff
@@ -369,6 +369,8 @@ bool TileManager::addTile(TileSet& tileSet, const TileID& _tileID) {
     if (tile) {
         m_tiles.push_back(tile);
 
+        // Update tile origin based on wrap (set in the new tileID)
+        tile->updateTileOrigin(_tileID.wrap);
```
Comment (blair1618) [inline core/src/tile/tileManager.cpp]: Is this call necessary? I don't see any effect when I comment it. It seems strange that this function would even be public; it seems that a `Tile` could potentially get an origin that doesn't correspond to the `wrap` value in its `TileID`.

**Comment by blair1618** [comment]
Hit a segfault when zooming out very rapidly, crash report: http://pastebin.com/UHbqCksZ

**Comment by tallytalwar** [comment]
Checking for the segfault issue.

File: core/src/tile/tileManager.cpp (hunk 1 above)
Comment (tallytalwar) [inline core/src/tile/tileManager.cpp]: Yes. Its when a tile data (corresponding to some xyz tileID) has been cached, and a new tile is being added with the wrapped IDs same as the cached tile's IDs. In this case you would want to have the tile grab the data from the cache and update the tile origin with the new wrap.

**Comment by hjanetzek** [comment]
Which reminds me that we need better error handling in mvt sources pbf parser - at least the runtime_error should be catched there :)

**Comment by hjanetzek** [comment]
Could it be that a tile is loaded twice, the first tile is added to the raw cache and starts processing - then the second comes in and is added to the cache overwriting the first that is currently processed by mvt parser?

**Comment by tallytalwar** [comment]
@hjanetzek thats very very possible.

**Comment by blair1618** [comment]
Turns out those crashes were totally my fault for making a mistake in the wrapping math and trying to fix nonexistent tiles - I think it's all corrected now.

**Pull request merged**